File: algorithms/trees/kd-tree/java/Node.java
public class Node {
    private int[] point;
    private int axis;
    private Node left;
    private Node right;

    public Node(int[] point, int axis) {
        this.point = point;
        this.axis = axis;
        this.left = null;
        this.right = null;
    }

    public int[] getPoint() {
        return point;
    }

    public int getAxis() {
        return axis;
    }

    public Node getLeft() {
        return left;
    }

    public Node getRight() {
        return right;
    }

    public void setLeft(Node left) {
        this.left = left;
    }

    public void setRight(Node right) {
        this.right = right;
    }
}

File: algorithms/trees/kd-tree/cpp/Node.cpp
#include "Node.h"

Node::Node(std::vector<int> point, int axis) : point(point), axis(axis), left(nullptr), right(nullptr) {}

std::vector<int> Node::getPoint() {
    return point;
}

int Node::getAxis() {
    return axis;
}

Node* Node::getLeft() {
    return left;
}

Node* Node::getRight() {
    return right;
}

void Node::setLeft(Node* left) {
    this->left = left;
}

void Node::setRight(Node* right) {
    this->right = right;
}
